### PR TITLE
QA tests

### PR DESCRIPTION
**Pull request by KimMcAlpine**
Unit tests for the new QA outputs

### CODE REVIEW

**Review by mauch — COMMENTED** [review]
I see that you are mimicking an image that is output by the imager and checking that the QA and PB code runs on it. I was just wondering if it might be a better test to actually have the imager generate the image from simulated visibilities instead. This way you are testing that any future changes to the output of the imager (eg. a FITS header keyword might change) won't break the QA code. Some simple examples of how to do this are in `test_continuum_pipeline.py`.
I thought I'd ask about this before looking more closely - there might be a good reason not to do it the way I'm suggesting that I'm missing.

**Comment by KimMcAlpine** [comment]
> I see that you are mimicking an image that is output by the imager and checking that the QA and PB code runs on it. I was just wondering if it might be a better test to actually have the imager generate the image from simulated visibilities instead. This way you are testing that any future changes to the output of the imager (eg. a FITS header keyword might change) won't break the QA code. Some simple examples of how to do this are in `test_continuum_pipeline.py`.
> I thought I'd ask about this before looking more closely - there might be a good reason not to do it the way I'm suggesting that I'm missing.

I am happy to do it the way your suggesting, my only reservation is that then your making the your test outcome very dependent on the success/failure of all that machinery that creates the images. So I feel that currently its a bit more self contained and a more direct test of just the things that are happening in the functions within the QA module. 
An alternative approach to solve the problem of changing keys in the image header is just to write some kind of test or schema that checks for any required keys, this way you can apply this test to the outputs of the imager (in the continuum pipeline tests), and similarly use it to validate the header keys I'm creating in my tests before  running the remaining tests. This could make it more obvious when key changes are the cause of failures in any future functions that use the images. 
Let me know what you think

**Comment by mauch** [comment]
I like that suggestion. There is also a related addition to the tests of
similarly checking the metadata output as well.
This can all be in a separate PR so I'll just go ahead and look at this as
is.

**Review by mauch — APPROVED** [review]
Looks good. Just a minor suggestion about HDU construction.

File: katacomb/katacomb/tests/test_qa.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+class TestMakePBImages:
+    def setup(self):
+        # Create temporary directory
+        self.tmpdir_in = tempfile.TemporaryDirectory()
+        in_array = np.ones([1, 3, 100, 100])
+        hdu = fits.PrimaryHDU(in_array)
+
+        for h in HDR_KEYS.keys():
+            hdu.header[h] = HDR_KEYS[h]
```
[the remark this answers is not in the document]
Comment (KimMcAlpine) [reply katacomb/katacomb/tests/test_qa.py:106]: done

File: katacomb/katacomb/tests/test_qa.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        gaus_data = gaus_mod(y, x)
+
+        # Leave in extra axes as QA code will try to drop extra axes
+        # inferred from the header
+        in_array += gaus_data[np.newaxis, np.newaxis, :, :]
+
+        hdu = fits.PrimaryHDU(in_array)
+
+        for h in HDR_KEYS.keys():
+            hdu.header[h] = HDR_KEYS[h]
```
[the remark this answers is not in the document]
Comment (KimMcAlpine) [reply katacomb/katacomb/tests/test_qa.py:163]: done

**Pull request merged**